### PR TITLE
Docs/leaner comments

### LINKED ISSUE #81: dev: clean up comments as discussed with the team
From @enitrat 
imo we go from 
> /// ADDMOD operation.
    /// Addition modulo operation
    /// # Additional informations:
    /// - Since:  Frontier
    /// - Group: Stop and Arithmetic Operations
    /// - Gas: 8
    /// - Stack consumed elements: 2
    /// - Stack produced elements: 1
    /// # Arguments
    /// * self - the execution context
    fn exec_addmod(ref self: ExecutionContext) {

to

> /// MULMOD operation.
    /// (a * b) % N: integer result of the multiplication followed by a modulo.
    /// All intermediate calculations of this operation are not subject to the 2^256 modulo.
    /// If the denominator is 0, the result will be 0.
    fn exec_mulmod(ref self: ExecutionContext) {

### PR DESCRIPTION
**Pull request by Eikix**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

resolves #81

### CODE REVIEW

File: src/instructions/stop_and_arithmetic_operations.cairo [hunk 1]
```diff
@@ -12,198 +12,113 @@ use kakarot::utils::math::{Exponentiation, ExponentiationModulo};
...
-    /// - Since:  Frontier
-    /// - Group: Stop and Arithmetic Operations
-    /// - Gas: 8
-    /// - Stack consumed elements: 2
-    /// - Stack produced elements: 1
-    /// # Arguments
-    /// * `self` - the execution context
+    /// 0x08 - ADDMOD
+    /// Addition and modulo operation
+    /// a % b: integer result of the signed integer modulo. If the denominator is 0, the result will be 0.
```
Comment (enitrat) [inline src/instructions/stop_and_arithmetic_operations.cairo:120]:
```suggestion
    /// (a + b) % N: integer result of the addition followed by a modulo. If the denominator is 0, the result will be 0.
```

File: src/context.cairo [hunk 2]
```diff
@@ -152,23 +152,13 @@ impl ExecutionContextImpl of ExecutionContextTrait {
...
         self.reverted = true;
         ArrayExtensionTrait::concat(ref self.return_data, revert_reason);
     }
 
     /// Reads and return data from bytecode.
     /// The program counter is incremented accordingly.
-    /// 
-    /// # Arguments
-    /// 
-    /// * `self` - The `ExecutionContext` instance to read the data from.
```
Comment (enitrat) [inline src/context.cairo:170]: I would keep this one

File: src/memory.cairo [hunk 3]
```diff
@@ -82,12 +72,6 @@ impl MemoryImpl of MemoryTrait {
     /// stored are within the same word in memory using the `_store_bytes_in_single_chunk` function. If the bytes
     /// span multiple words, the function stores the first word using the `_store_first_word` function, the aligned
     /// words using the `_store_aligned_words` function, and the last word using the `_store_last_word` function.
-    ///
-    /// # Arguments
-    ///
-    /// * `self` - A mutable reference to the `Memory` instance to store the bytes in.
-    /// * `elements` - A span of bytes to store in memory.
-    /// * `offset` - The offset within memory to store the bytes at.
```
Comment (enitrat) [inline src/memory.cairo:90]: I would also keep these comments, as the operations in these functions are not super simple.

**Review by enitrat — COMMENTED** [review]
You will also need to update the changelot

File: src/instructions/stop_and_arithmetic_operations.cairo [hunk 4]
```diff
@@ -12,198 +12,113 @@ use kakarot::utils::math::{Exponentiation, ExponentiationModulo};
...
-    /// - Group: Stop and Arithmetic Operations
-    /// - Gas: 5
-    /// - Stack consumed elements: 2
-    /// - Stack produced elements: 1
-    /// # Arguments
-    ///TODO
-    /// - Implement me.
-    /// * `self` - the execution context
+    /// a / b: integer result of the signed integer division. 
+    /// If the denominator is 0, the result will be 0.
```
[the remark this answers is not in the document]
Comment (enitrat) [reply src/instructions/stop_and_arithmetic_operations.cairo:82]: #89

File: src/instructions/duplication_operations.cairo [hunk 5]
```diff
@@ -4,256 +4,95 @@
...
 
-///Generic DUP operation
-/// Duplicate the top i-th stack item to the top of the stack.
-/// # Arguments
-/// * `ctx` - the execution context
-/// * `i` - the i-th stack item to duplicate
-/// # TODO
-/// - Implement me.
-fn exec_dup_i(ref context: ExecutionContext, i: u8) {}
+mod internal {
```
[the remark this answers is not in the document]
Comment (Eikix) [reply src/instructions/duplication_operations.cairo:7]: i agree, but might create some confusion? we can try both approaches and see which is most intuitive

**Review by enitrat — APPROVED** [review]

File: src/instructions/block_information.cairo [hunk 6]
```diff
@@ -4,129 +4,46 @@
...
-/// - Group: Block Information
-/// - Gas: 2
-/// - Stack consumed elements: 0
-/// - Stack produced elements: 1
-/// # Arguments
-/// * `ctx` - the execution context
-/// # TODO
-/// - Implement me.
-fn exec_difficulty(ref context: ExecutionContext) {}
+/// 0x44 - PREVRANDAO
```
Comment (greged93) [inline src/instructions/block_information.cairo:27]: Would add
```
// Get the previous block’s RANDAO mix
```
to stay in line with the rest of the docs

File: src/instructions/memory_operations.cairo [hunk 7]
```diff
@@ -6,159 +6,61 @@ use kakarot::context::ExecutionContextTrait;
 
 /// MLOAD operation.
```
Comment (greged93) [inline src/instructions/memory_operations.cairo:7]: Opcode number missing

File: src/instructions/system_operations.cairo [hunk 8]
```diff
@@ -4,133 +4,43 @@
 use kakarot::context::ExecutionContext;
```
Comment (greged93) [inline src/instructions/system_operations.cairo:4]: No one liner explanation ?

File: src/instructions/comparison_operations.cairo [hunk 9]
```diff
@@ -1,201 +1,63 @@
-//! Comparison & Bitwise Logic Operations
```
Comment (greged93) [inline src/instructions/comparison_operations.cairo:1]: No one liner explanation?

**Pull request merged**